src/main/java/com/scalyr/integrations/kafka/AddEventsClient.java

```diff
@@ -170,6 +187,16 @@ public AddEventsClient(String scalyrUrl, String apiKey, long addEventsTimeoutMs,
    */
   private AddEventsResponse addEventsWithRetry(byte[] addEventsPayload, long startTimeMs) {
     log.debug("addEvents payload size {} bytes", addEventsPayload.length);
+
+    // 6 MB add events payload exceeded.  Log the issue and skip this message.
+    if (addEventsPayload.length > MAX_ADD_EVENTS_PAYLOAD_BYTES) {
```

Comment:
Hm, I just a had a look at the code and this operates on an already serialized and compressed data, right?

And looking at the server side code, the limit appears to be set when we are reading compressed body into buffer used for uncompressed data aka the limit appears to be on the uncompressed body size (https://github.com/scalyr/scalyr/blob/master/ScalyrSite/src/com/scalyr/config/api/GenericApiServlet.java#L534).

So the solution might be to just check the size of serialized data before compressing it...